cognite_toolkit/cdf_tk/load.py

```diff
@@ -835,11 +910,237 @@ def create(
                     print(f"  [bold yellow]WARNING:[/] File {meta.external_id} already exists, skipping upload.")
             except Exception as e:
                 print(f"[bold red]ERROR:[/] Failed to upload file {datafile.name}.\n{e}")
-                ToolGlobals.failed = True
+                self.ToolGlobals.failed = True
                 return created
         return created
 
 
+@final
+class SpaceLoader(Loader[str, SpaceApply, SpaceApplyList]):
+    api_name = "data_modeling.spaces"
+    folder_name = "data_models"
+    filename_pattern = r"^.*\.?(space)$"
+    resource_cls = SpaceApply
+    list_cls = SpaceApplyList
+
+    @classmethod
+    def get_required_capability(cls, ToolGlobals: CDFToolConfig) -> list[Capability]:
+        return [
+            DataModelsAcl(
+                [DataModelsAcl.Action.Read, DataModelsAcl.Action.Write],
+                DataModelsAcl.Scope.All(),
+            ),
+            # Needed to delete instances
+            DataModelInstancesAcl(
+                [DataModelInstancesAcl.Action.Read, DataModelInstancesAcl.Action.Write],
+                DataModelInstancesAcl.Scope.All(),
+            ),
+        ]
+
+    @classmethod
+    def get_id(cls, item: SpaceApply) -> str:
+        return item.space
+
+    def delete(self, ids: Sequence[str], drop_data: bool) -> int:
+        if not drop_data:
+            print("  [bold]INFO:[/] Skipping deletion of spaces as drop_data flag is not set...")
+            return 0
+        print("[bold]Deleting existing data...[/]")
+        for space in ids:
```

Comment:
Yes, that is checked on the line above, the if-statement before: `print("  [bold]INFO:[/] Skipping deletion of spaces as drop_data flag is not set...")`.

In the dependencies, you see that space <-containers <- nodes <- edges, so when you call this for delete, it will go in the opposite direction. We have to then delete the extra instances that are in the space, which are not governed by the toolkit.

This requires that we remove `drop_data` from `deploy` as we discussed before. Will do it in this PR.